src/session.rs

```diff
@@ -266,26 +280,31 @@ where
             let mut response = inner.call(request).await?;
 
             if session.is_destroyed() {
-                session_layer
-                    .store
-                    .destroy_session(session)
-                    .await
-                    .expect("Could not destroy session.");
+                if let Err(e) = session_layer.store.destroy_session(session).await {
+                    tracing::error!("Failed to destroy session: {:?}", e);
+                    *response.status_mut() = StatusCode::INTERNAL_SERVER_ERROR;
+                }
+
+                let removal_cookie = session_layer.build_removal_cookie(session_layer.secure);
+
+                response.headers_mut().insert(
+                    SET_COOKIE,
+                    HeaderValue::from_str(&removal_cookie.to_string()).unwrap(),
+                );
             } else if session_layer.save_unchanged || session.data_changed() {
-                let cookie = session_layer
-                    .store
-                    .store_session(session)
-                    .await
-                    .expect("Could not store session.")
-                    .map(|cookie_value| {
-                        session_layer.build_cookie(session_layer.secure, cookie_value)
-                    });
-
-                if let Some(cookie) = cookie {
-                    response.headers_mut().insert(
-                        SET_COOKIE,
-                        HeaderValue::from_str(&cookie.to_string()).unwrap(),
-                    );
+                match session_layer.store.store_session(session).await {
+                    Ok(Some(cookie_value)) => {
+                        let cookie = session_layer.build_cookie(session_layer.secure, cookie_value);
+                        response.headers_mut().insert(
+                            SET_COOKIE,
+                            HeaderValue::from_str(&cookie.to_string()).unwrap(),
+                        );
+                    }
+                    Ok(None) => {}
```

Comment:
I'm not sure how to handle this branch. I suppose due to our setup, this basically never happens.

Reply:
I wonder we could reach this branch if we were to clear the storage (perhaps manually, as an example) while clients still hold cookies. If that's a valid case, would it make sense to send the removal cookie here too?